src/Common/HashTable/HashTable.h

```diff
@@ -850,6 +786,69 @@ class HashTable :
 
 
 public:
+    /// Increase the size of the buffer.
+    void resize(size_t for_num_elems = 0, size_t for_buf_size = 0)
```

Comment:
I think it is better to keep `resize` private (also, it would be good to save blame for it).
Just add public `reserve(num_elements)` method, the same as for `std::vector` or `PODArray`

Reply:
>also, it would be good to save blame for it)

Great point!